### PR TITLE
Add unit test for generate_cert.go

### PR DESCRIPTION
**Pull request by wlu2016**
@lookuptable

### CODE REVIEW

**Comment by lookuptable** [comment]
This closes #30 .

File: certmanager/generate_cert_test.go [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+	"time"
+)
+
+// VerifyFields contains the certficate fields to verify in the test.
+type VerifyFields struct {
+	notBefore time.Time
+	notAfter  time.Time
+	extKeyUsage x509.ExtKeyUsage
+	isCA		bool
+	org		string
```
Comment (lookuptable) [inline certmanager/generate_cert_test.go]: These violate the Go format. Please run `go fmt ./...` to have them auto-formatted.

File: certmanager/generate_cert_test.go [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+		ValidFrom: nowInString,
+		ValidFor: time.Hour * 24 * 30,
+		SignerCert: nil,
+		SignerPriv: nil,
+		Org: "MyOrg",
+		IsCA: true,
+		IsSelfSigned: true,
+		IsClient: false,
+	}
+	notBefore, _ := time.Parse("Jan 2 15:04:05 2006", nowInString)
```
Comment (lookuptable) [inline certmanager/generate_cert_test.go]: Consider making the layout string a constant so you can just use it here and in the `GenCert()` method.

File: certmanager/generate_cert_test.go [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+
+	block, _ = pem.Decode(privPem)
+	if block == nil || block.Type != "RSA PRIVATE KEY" {
+		t.Errorf("failed to decode PEM block containing private key")
+	}
+	priv, err := x509.ParsePKCS1PrivateKey(block.Bytes)
+	if err != nil {
+		t.Errorf("failed to parse private key: " + err.Error())
+	}
+	if !reflect.DeepEqual(priv.PublicKey, *cert.PublicKey.(*rsa.PublicKey)) {
```
Comment (lookuptable) [inline certmanager/generate_cert_test.go]: You probably don't need to type conversion for `cert.PublicKey`. I think `reflect.DeepEqual()` is smart enough to detect that.

File: certmanager/generate_cert_test.go [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+		notBefore: cert.NotBefore,
+		notAfter: cert.NotAfter,
+		extKeyUsage: cert.ExtKeyUsage[0],
+		isCA: cert.IsCA,
+		org: cert.Issuer.Organization[0],
+	}
+	if !reflect.DeepEqual(expectedFields, certFields) {
+		t.Errorf("{notBefore, notAfter, extKeyUsage, isCA, org}:")
+		t.Errorf("expected: %s", expectedFields)
+		t.Errorf("actual: %s", certFields)
```
Comment (lookuptable) [inline certmanager/generate_cert_test.go]: Line 218 and 219 are dead code, The test exits after the first `t.Errorf()`.

File: certmanager/generate_cert_test.go (hunk 1 above)
Comment (wlu2016) [inline certmanager/generate_cert_test.go]: done

File: certmanager/generate_cert_test.go (hunk 2 above)
Comment (wlu2016) [inline certmanager/generate_cert_test.go]: good point, done

File: certmanager/generate_cert_test.go (hunk 3 above)
Comment (wlu2016) [inline certmanager/generate_cert_test.go]: it is not that smart :( which is why I did that type conversion

File: certmanager/generate_cert_test.go (hunk 4 above)
Comment (wlu2016) [inline certmanager/generate_cert_test.go]: actually, they do get executed as I tested.

**Comment by lookuptable** [comment]
There's still a few linter violations. You can see details here: https://travis-ci.org/istio/auth/builds/209146853 around line 610.

**Comment by wlu2016** [comment]
fixed linter errors, PTAL

**Pull request merged**